### DISCUSSION TITLE
Errored Counts decrease when messages are removed from channels

### DISCUSSION THREAD

**Discussion by rp-ryanwilson** [nextgenhealthcare/connect#5684] [Feature/Enhancement Ideas]
Currently when we remove a message from a channel that has erred it does not decrease the 'Errored' count reflected on the dashboard.  It only decreases it if we reprocess the error and it does not error the second time.  This often times leaves us with an inaccurate count of errors reflected on the dashboard for channels.

**Comment by pacmano1** [comment]
If you are asking for a feature to make the behavior different, add a detailed feature request to the issues section.

**Reply by pacmano1** [reply]
Well, your feature request might work for you but perhaps not others.  

So maybe take the time to suggest thiings like "add option to update statistics for removed message" and be considerate of lifetime stats vs current stats?   Also, the feature request has a template, it is usually best to follow it.

**Reply by rp-ryanwilson** [reply]
Thanks for the feedback, I added a feature request to the issues section as advised and followed the template.

**Comment by kirbykn2** [comment]
Seems like a process issue.  If you are removing messages and not resolving
the issue, remove all error'd messages then reset the error count.

On Thu, Feb 16, 2023 at 12:01 PM pacmano1 ***@***.***> wrote:

> Well, your feature request might work for you but perhaps not others.
>
> So maybe take the time to suggest thiings like "add option to update
> statistics for removed message" and be considerate of lifetime stats vs
> current status? Also, the feature request has a template, it is usually
> best to follow it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/5684#discussioncomment-4998744>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWDZ3QDKPOOYD6CYNG3TWXZMPPANCNFSM6AAAAAAU6MVHCQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <nextgenhealthcare/connect/repo-discussions/5684/comments/4998744@
> github.com>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by pacmano1** [reply]
@kirbykn2 I agree.  But it's probably up to the user to define what should remain as an error from a stats perspective.   For example, I might have mistyped an API endpoint and don't really want my stupidity to show up in the error count.

**Reply by rp-ryanwilson** [reply]
The problem is with the volume of messages it's not always that we can remove all the errors, they error for all kinds of different reasons.  Sometimes we remove the messages, sometimes we do not, so we can't reset the error count to 0.  When a message is removed it shouldn't still be counted in the total number of errors on a channel, I can't see how that's beneficial.  If you still need it to count in the total number of errors then don't remove it.

**Comment by kirbykn2** [comment]
It is beneficial to know the total number of unresolved error'd messages in
a channel.  I am with you, there is definitely room for some additional
options/configurations in this area to support different use cases.
Additionally, adding an option to decrement the error count when a message
is removed does make sense in certain cases.  But what about the other
counts?

On Thu, Feb 16, 2023 at 12:41 PM rp-ryanwilson ***@***.***>
wrote:

> The problem is with the volume of messages it's not always that we can
> remove all the errors, they error for all kinds of different reasons.
> Sometimes we remove the messages, sometimes we do not, so we can't reset
> the error count to 0. When a message is removed it shouldn't still be
> counted in the total number of errors on a channel, I can't see how that's
> beneficial. If you still need it to count in the total number of errors
> then don't remove it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/5684#discussioncomment-4999140>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD6CT2Q77UCRBFBHUZTWXZRGLANCNFSM6AAAAAAU6MVHCQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by rp-ryanwilson** [reply]
I don't think other counts matter as much because we still did receive or send the message, but when we remove it from the channel we need the errored count to decrease in order to track how many active errors exist on the channel.

If the option has to be to decrease all the counts when a message is removed then that would work for us.
